aodncore/util/misc.py

```diff
@@ -95,6 +96,22 @@ def format_exception(exception):
     return "{cls}: {message}".format(cls=exception.__class__.__name__, message=exception)
 
 
+def get_fields_from_str(str, pattern):
```

Comment:
str is a reserved word, so needs to be changed. something like "input_string". Apart from that, I can't think of a better more concise name, so happy to merge!

Reply:
get_pattern_subgroups_from_string?